src/canvas.class.js

```diff
@@ -492,6 +492,20 @@
      * @return {Boolean}
      */
     isTargetTransparent: function (target, x, y) {
+      if (target.shouldCache && target._cacheCanvas && !target.isCacheDirty()) {
+        var center = target.getCenterPoint(),
+            curRelTargetX = x - center.x,
+            curRelTargetY = y - center.y;
+
+        curRelTargetX = (target._cacheCanvas.width / 2) + (curRelTargetX / target.zoomX);
+        curRelTargetY = (target._cacheCanvas.height / 2) + (curRelTargetY / target.zoomY);
```

Comment:
for blurriness reasons more than width/2 there is another property where we can see the center of the obect on the canvas, i think is called cacheTranslateX/Y or something like that, you find it in UpdateCacheCanvas function.

Also the translation respect the center should take care of the object rotation using the matrices directly.

Reply:
Okay, I tried using cacheTranslateX/Y in various ways, but I was getting weird results. I'll give it another try though.